src/channels/views/ChannelsList/ChannelsList.tsx

```diff
@@ -14,23 +18,54 @@ interface ChannelsListProps {
   params: ChannelsListUrlQueryParams;
 }
 
-export const ChannelsList: React.FC<ChannelsListProps> = () => {
+export const ChannelsList: React.FC<ChannelsListProps> = ({ params }) => {
   const navigate = useNavigator();
   const { data, loading } = useChannelsList({ displayLoader: true });
 
+  const channelsChoices = useMemo(
+    () =>
+      params.id &&
+      data?.channels
+        ?.map(channel => ({
+          label: channel.name,
+          value: channel.id
+        }))
+        .filter(channel => channel.value !== params.id),
+    [data?.channels?.length, params.id]
```

Comment:
I'm not sure if depending on arrays length is good enough - let's say that one channel got updated, but length is still the same. Choices are not updated.
TBH I don't see any reason for this to be memoized, creating this array is, well, free.